### PR TITLE
Update Novoalign to V4.03.04

### PR DESCRIPTION
**Pull request by colinhercus**
Describe your pull request here

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

For members of the Bioconda project, the following command is also available:

<table>
  <tr>
    <td><code>@BiocondaBot please merge</code></td>
    <td>Upload built packages/containers and merge a PR. <br />Someone must approve a PR first! <br />This reduces CI build time by reusing built artifacts.</td>
  </tr>
</table>

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

**Comment by mbargull** [comment]
For context, the small container base image only has the (partially modified) Debian packages
```sh
# podman run --rm quay.io/bioconda/base-glibc-busybox-bash cat /.pkg.lst
gcc-8-base
libc6
libgcc1
bash
libc-bin
libtinfo6
ncurses-base
base-files
```
and uses Busybox instead of GNU Coreutils.
On the other side, the `extended-base` image is based on upstream `docker.io/debian:10-slim` and has a lot more packages:
```sh
# podman run --rm --env=PAGER=cat quay.io/bioconda/base-glibc-debian-bash dpkg --list | grep \^ii
ii  adduser                 3.118                  all          add and remove users and groups
ii  apt                     1.8.2.2                amd64        commandline package manager
ii  base-files              10.3+deb10u9           amd64        Debian base system miscellaneous files
ii  base-passwd             3.5.46                 amd64        Debian base system master password and group files
ii  bash                    5.0-4                  amd64        GNU Bourne Again SHell
ii  bsdutils                1:2.33.1-0.1           amd64        basic utilities from 4.4BSD-Lite
ii  coreutils               8.30-3                 amd64        GNU core utilities
ii  dash                    0.5.10.2-5             amd64        POSIX-compliant shell
ii  debconf                 1.5.71                 all          Debian configuration management system
ii  debian-archive-keyring  2019.1+deb10u1         all          GnuPG archive keys of the Debian archive
ii  debianutils             4.8.6.1                amd64        Miscellaneous utilities specific to Debian
ii  diffutils               1:3.7-3                amd64        File comparison utilities
ii  dpkg                    1.19.7                 amd64        Debian package management system
ii  e2fsprogs               1.44.5-1+deb10u3       amd64        ext2/ext3/ext4 file system utilities
ii  fdisk                   2.33.1-0.1             amd64        collection of partitioning utilities
ii  findutils               4.6.0+git+20190209-2   amd64        utilities for finding files--find, xargs
ii  gcc-8-base:amd64        8.3.0-6                amd64        GCC, the GNU Compiler Collection (base package)
ii  gpgv                    2.2.12-1+deb10u1       amd64        GNU privacy guard - signature verification tool
ii  grep                    3.3-1                  amd64        GNU grep, egrep and fgrep
ii  gzip                    1.9-3                  amd64        GNU compression utilities
ii  hostname                3.21                   amd64        utility to set/show the host name or domain name
ii  init-system-helpers     1.56+nmu1              all          helper tools for all init systems
ii  libacl1:amd64           2.2.53-4               amd64        access control list - shared library
ii  libapt-pkg5.0:amd64     1.8.2.2                amd64        package management runtime library
ii  libattr1:amd64          1:2.4.48-4             amd64        extended attribute handling - shared library
ii  libaudit-common         1:2.8.4-3              all          Dynamic library for security auditing - common files
ii  libaudit1:amd64         1:2.8.4-3              amd64        Dynamic library for security auditing
ii  libblkid1:amd64         2.33.1-0.1             amd64        block device ID library
ii  libbsd0:amd64           0.9.1-2+deb10u1        amd64        utility functions from BSD systems - shared library
ii  libbz2-1.0:amd64        1.0.6-9.2~deb10u1      amd64        high-quality block-sorting file compressor library - runtime
ii  libc-bin                2.28-10                amd64        GNU C Library: Binaries
ii  libc6:amd64             2.28-10                amd64        GNU C Library: Shared libraries
ii  libcap-ng0:amd64        0.7.9-2                amd64        An alternate POSIX capabilities library
ii  libcom-err2:amd64       1.44.5-1+deb10u3       amd64        common error description library
ii  libdb5.3:amd64          5.3.28+dfsg1-0.5       amd64        Berkeley v5.3 Database Libraries [runtime]
ii  libdebconfclient0:amd64 0.249                  amd64        Debian Configuration Management System (C-implementation library)
ii  libdrm-common           2.4.97-1               all          Userspace interface to kernel DRM services -- common files
ii  libdrm2:amd64           2.4.97-1               amd64        Userspace interface to kernel DRM services -- runtime
ii  libedit2:amd64          3.1-20181209-1         amd64        BSD editline and history libraries
ii  libexpat1:amd64         2.2.6-2+deb10u1        amd64        XML parsing C library - runtime library
ii  libext2fs2:amd64        1.44.5-1+deb10u3       amd64        ext2/ext3/ext4 file system libraries
ii  libfdisk1:amd64         2.33.1-0.1             amd64        fdisk partitioning library
ii  libffi6:amd64           3.2.1-9                amd64        Foreign Function Interface library runtime
ii  libgcc1:amd64           1:8.3.0-6              amd64        GCC support library
ii  libgcrypt20:amd64       1.8.4-5                amd64        LGPL Crypto library - runtime library
ii  libgl1:amd64            1.1.0-1                amd64        Vendor neutral GL dispatch library -- legacy GL support
rc  libgl1-mesa-dri:amd64   18.3.6-2+deb10u1       amd64        free implementation of the OpenGL API -- DRI modules
ii  libgl1-mesa-glx:amd64   18.3.6-2+deb10u1       amd64        transitional dummy package
ii  libglapi-mesa:amd64     18.3.6-2+deb10u1       amd64        free implementation of the GL API -- shared library
ii  libglvnd0:amd64         1.1.0-1                amd64        Vendor neutral GL dispatch library
ii  libglx-mesa0:amd64      18.3.6-2+deb10u1       amd64        free implementation of the OpenGL API -- GLX vendor library
ii  libglx0:amd64           1.1.0-1                amd64        Vendor neutral GL dispatch library -- GLX support
ii  libgmp10:amd64          2:6.1.2+dfsg-4         amd64        Multiprecision arithmetic library
ii  libgnutls30:amd64       3.6.7-4+deb10u6        amd64        GNU TLS library - main runtime library
ii  libgpg-error0:amd64     1.35-1                 amd64        GnuPG development runtime library
ii  libgssapi-krb5-2:amd64  1.17-3+deb10u1         amd64        MIT Kerberos runtime libraries - krb5 GSS-API Mechanism
ii  libhogweed4:amd64       3.4.1-1                amd64        low level cryptographic library (public-key cryptos)
ii  libidn2-0:amd64         2.0.5-1+deb10u1        amd64        Internationalized domain names (IDNA2008/TR46) library
ii  libk5crypto3:amd64      1.17-3+deb10u1         amd64        MIT Kerberos runtime libraries - Crypto Library
ii  libkeyutils1:amd64      1.6-6                  amd64        Linux Key Management Utilities (library)
ii  libkrb5-3:amd64         1.17-3+deb10u1         amd64        MIT Kerberos runtime libraries
ii  libkrb5support0:amd64   1.17-3+deb10u1         amd64        MIT Kerberos runtime libraries - Support library
ii  liblz4-1:amd64          1.8.3-1                amd64        Fast LZ compression algorithm library - runtime
ii  liblzma5:amd64          5.2.4-1                amd64        XZ-format compression library
ii  libmount1:amd64         2.33.1-0.1             amd64        device mounting library
ii  libncurses6:amd64       6.1+20181013-2+deb10u2 amd64        shared libraries for terminal handling
ii  libncursesw6:amd64      6.1+20181013-2+deb10u2 amd64        shared libraries for terminal handling (wide character support)
ii  libnettle6:amd64        3.4.1-1                amd64        low level cryptographic library (symmetric and one-way cryptos)
ii  libp11-kit0:amd64       0.23.15-2+deb10u1      amd64        library for loading and coordinating access to PKCS#11 modules - runtime
ii  libpam-modules:amd64    1.3.1-5                amd64        Pluggable Authentication Modules for PAM
ii  libpam-modules-bin      1.3.1-5                amd64        Pluggable Authentication Modules for PAM - helper binaries
ii  libpam-runtime          1.3.1-5                all          Runtime support for the PAM library
ii  libpam0g:amd64          1.3.1-5                amd64        Pluggable Authentication Modules library
ii  libpcre3:amd64          2:8.39-12              amd64        Old Perl 5 Compatible Regular Expression Library - runtime files
ii  libprocps7:amd64        2:3.3.15-2             amd64        library for accessing process information from /proc
ii  libseccomp2:amd64       2.3.3-4                amd64        high level interface to Linux seccomp filter
ii  libselinux1:amd64       2.8-1+b1               amd64        SELinux runtime shared libraries
ii  libsemanage-common      2.8-2                  all          Common files for SELinux policy management libraries
ii  libsemanage1:amd64      2.8-2                  amd64        SELinux policy management library
rc  libsensors-config       1:3.5.0-3              all          lm-sensors configuration files
ii  libsepol1:amd64         2.8-1                  amd64        SELinux library for manipulating binary security policies
ii  libsmartcols1:amd64     2.33.1-0.1             amd64        smart column output alignment library
ii  libss2:amd64            1.44.5-1+deb10u3       amd64        command-line interface parsing library
ii  libssl1.1:amd64         1.1.1d-0+deb10u6       amd64        Secure Sockets Layer toolkit - shared libraries
ii  libstdc++6:amd64        8.3.0-6                amd64        GNU Standard C++ Library v3
ii  libsystemd0:amd64       241-7~deb10u7          amd64        systemd utility library
ii  libtasn1-6:amd64        4.13-3                 amd64        Manage ASN.1 structures (runtime)
ii  libtinfo6:amd64         6.1+20181013-2+deb10u2 amd64        shared low-level terminfo library for terminal handling
ii  libudev1:amd64          241-7~deb10u7          amd64        libudev shared library
ii  libunistring2:amd64     0.9.10-1               amd64        Unicode string library for C
ii  libuuid1:amd64          2.33.1-0.1             amd64        Universally Unique ID library
ii  libx11-6:amd64          2:1.6.7-1+deb10u1      amd64        X11 client-side library
ii  libx11-data             2:1.6.7-1+deb10u1      all          X11 client-side library
ii  libx11-xcb1:amd64       2:1.6.7-1+deb10u1      amd64        Xlib/XCB interface library
ii  libxau6:amd64           1:1.0.8-1+b2           amd64        X11 authorisation library
ii  libxcb-dri2-0:amd64     1.13.1-2               amd64        X C Binding, dri2 extension
ii  libxcb-dri3-0:amd64     1.13.1-2               amd64        X C Binding, dri3 extension
ii  libxcb-glx0:amd64       1.13.1-2               amd64        X C Binding, glx extension
ii  libxcb-present0:amd64   1.13.1-2               amd64        X C Binding, present extension
ii  libxcb-sync1:amd64      1.13.1-2               amd64        X C Binding, sync extension
ii  libxcb1:amd64           1.13.1-2               amd64        X C Binding
ii  libxdamage1:amd64       1:1.1.4-3+b3           amd64        X11 damaged region extension library
ii  libxdmcp6:amd64         1:1.1.2-3              amd64        X11 Display Manager Control Protocol library
ii  libxext6:amd64          2:1.3.3-1+b2           amd64        X11 miscellaneous extension library
ii  libxfixes3:amd64        1:5.0.3-1              amd64        X11 miscellaneous 'fixes' extension library
ii  libxshmfence1:amd64     1.3-1                  amd64        X shared memory fences - shared library
ii  libxxf86vm1:amd64       1:1.1.4-1+b2           amd64        X11 XFree86 video mode extension library
ii  libzstd1:amd64          1.3.8+dfsg-3+deb10u2   amd64        fast lossless compression algorithm
rc  locales                 2.28-10                all          GNU C Library: National Language (locale) data [support]
ii  login                   1:4.5-1.1              amd64        system login tools
ii  lsb-base                10.2019051400          all          Linux Standard Base init script functionality
ii  mawk                    1.3.3-17+b3            amd64        a pattern scanning and text processing language
ii  mount                   2.33.1-0.1             amd64        tools for mounting and manipulating filesystems
ii  ncurses-base            6.1+20181013-2+deb10u2 all          basic terminal type definitions
ii  ncurses-bin             6.1+20181013-2+deb10u2 amd64        terminal-related programs and man pages
ii  openssh-client          1:7.9p1-10+deb10u2     amd64        secure shell (SSH) client, for secure access to remote machines
ii  passwd                  1:4.5-1.1              amd64        change and administer password and group data
ii  perl-base               5.28.1-6+deb10u1       amd64        minimal Perl system
ii  procps                  2:3.3.15-2             amd64        /proc file system utilities
ii  sed                     4.7-1                  amd64        GNU stream editor for filtering/transforming text
ii  sysvinit-utils          2.93-8                 amd64        System-V-like utilities
ii  tar                     1.30+dfsg-6            amd64        GNU version of the tar archiving utility
ii  tzdata                  2021a-0+deb10u1        all          time zone and daylight-saving time data
ii  util-linux              2.33.1-0.1             amd64        miscellaneous system utilities
ii  zlib1g:amd64            1:1.2.11.dfsg-1        amd64        compression library - runtime
```
The pre-built binaries apparently need some things not available in the smaller base image.

File: recipes/novoalign/meta.yaml [hunk 1]
```diff
@@ -26,6 +26,8 @@ requirements:
   run:
     # curl is needed to access Novocraft's website during package install -- see post-link.sh
     - curl
+    # Novoalign 4.03.04 pre-compiled binary needs libsmartcols from util-linux.
+    - util-linux
```
Comment (mbargull) [inline recipes/novoalign/meta.yaml:30]: `strace -f -e openat novoalign --version 2>&1 | grep -F .so`
includes
`[pid 393] openat(AT_FDCWD, "/lib/x86_64-linux-gnu/libsmartcols.so.1", O_RDONLY|O_CLOEXEC) = 3`
which means it needs `libsmartcols` which is part of the `util-linux` package.

Comment (mbargull) [reply recipes/novoalign/meta.yaml:30]: And if we look at the `strace` lines before the above one, we can see that it comes from `lscpu` (also from `util-linux`):
```sh
# strace -f novoalign --version 2>&1 | grep -FB10 .so | grep -A10 exec
[pid   704] execve("/bin/sh", ["sh", "-c", "lscpu -ap 2>/dev/null"], 0x7fff32e27480 /* 14 vars */ <unfinished ...>
[pid   703] read(3,  <unfinished ...>
[pid   704] <... execve resumed> )      = 0
[pid   704] brk(NULL)                   = 0x55c086eda000
[pid   704] access("/etc/ld.so.preload", R_OK) = -1 ENOENT (No such file or directory)
[pid   704] openat(AT_FDCWD, "/etc/ld.so.cache", O_RDONLY|O_CLOEXEC) = 3
[pid   704] fstat(3, {st_mode=S_IFREG|0644, st_size=16451, ...}) = 0
[pid   704] mmap(NULL, 16451, PROT_READ, MAP_PRIVATE, 3, 0) = 0x7fc13b466000
[pid   704] close(3)                    = 0
[pid   704] openat(AT_FDCWD, "/lib/x86_64-linux-gnu/libc.so.6", O_RDONLY|O_CLOEXEC) = 3
--
[pid   704] stat("/usr/local/bin/lscpu", 0x7ffc18dd71e0) = -1 ENOENT (No such file or directory)
[pid   704] stat("/usr/local/sbin/lscpu", 0x7ffc18dd71e0) = -1 ENOENT (No such file or directory)
--
[pid   705] execve("/usr/bin/lscpu", ["lscpu", "-ap"], 0x55c086d30c18 /* 14 vars */) = 0
[pid   705] brk(NULL)                   = 0x55d55b7e7000
[pid   705] access("/etc/ld.so.preload", R_OK) = -1 ENOENT (No such file or directory)
[pid   705] openat(AT_FDCWD, "/etc/ld.so.cache", O_RDONLY|O_CLOEXEC) = 3
[pid   705] fstat(3, {st_mode=S_IFREG|0644, st_size=16451, ...}) = 0
[pid   705] mmap(NULL, 16451, PROT_READ, MAP_PRIVATE, 3, 0) = 0x7f6fd728d000
[pid   705] close(3)                    = 0
[pid   705] openat(AT_FDCWD, "/lib/x86_64-linux-gnu/libsmartcols.so.1", O_RDONLY|O_CLOEXEC) = 3
[pid   705] read(3, "\177ELF\2\1\1\0\0\0\0\0\0\0\0\0\3\0>\0\1\0\0\0P\217\0\0\0\0\0\0"..., 832) = 832
[pid   705] fstat(3, {st_mode=S_IFREG|0644, st_size=223360, ...}) = 0
[pid   705] mmap(NULL, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x7f6fd728b000
```

**Comment by mbargull** [comment]
@colinhercus, I was looking for a way to submit a bug report an found http://www.novocraft.com/forums/forum/support-2/ . And there I saw that you are affiliated with Novocraft :).
So, since I'm talking to the "source" directly here, I'm taking the liberty to omit the forum registration and post directly here:
Could you add a check for `lscpu`'s output and show an error message if it fails to execute to avoid the observed segmentation fault?

**Review by mbargull — APPROVED** [review]
LGTM -- but there are a bunch of my own changes I'd be approving :D.
@colinhercus or others, do the added changes look good to you too?

**Comment by colinhercus** [comment]
@mbargull  I don't directly call lscpu. It must be coming out of some other routine we call. We do check how many cores the server has using sysconf(_SC_NPROCESSORS_ONLN) and how many we have affinity for using TThreadPool::numAffinity(); so that we can create just enough threads for one per core. 
I'm okay with any changes you make that will get the Pull request through. Thanks for your help.

**Comment by mbargull** [comment]
> @mbargull I don't directly call lscpu. It must be coming out of some other routine we call. We do check how many cores the server has using sysconf(_SC_NPROCESSORS_ONLN) and how many we have affinity for using TThreadPool::numAffinity(); so that we can create just enough threads for one per core.

I don't know from where `TThreadPool::numAffinity()` comes from and what it (or other routines you call) might invoke, so I can't give any further pointers there. But, acc. to the strace, the `sh -c 'lscpu -ap 2>/dev/null'` happens somewhere between `sysconf(_SC_NPROCESSORS_ONLN)` (since the trace shows reading from `/sys/devices/system/cpu/online`) and the `sched_getaffinity`/`sched_setaffinity` calls afterwards.

> I'm okay with any changes you make that will get the Pull request through. Thanks for your help.

Alright, then let's merge!

**Pull request merged**